### PR TITLE
[IPU] update ipu releated passes p0

### PR DESCRIPTION
**Pull request by gglin001**
### PR types

Others
### PR changes

Others
### Describe

ipu releated updates

check p1 in https://github.com/PaddlePaddle/Paddle/pull/38985

### CODE REVIEW

File: paddle/fluid/framework/ir/ipu/ipu_pass_base.cc [hunk 1]
```diff
@@ -18,10 +18,7 @@ namespace paddle {
 namespace framework {
 namespace ir {
 
-void IPUPassBase::Init(const std::string& repr, Graph* graph) const {
-  repr_ = repr;
-  graph_ = graph;
-}
+// pass
```
Comment (qili93) [inline paddle/fluid/framework/ir/ipu/ipu_pass_base.cc]: ipu_pass_base.cc / ipu_pass_base.h都是空文件，建议删掉

File: paddle/fluid/framework/ir/ipu/transfer_cast_op_pass.cc [hunk 2]
```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
```
Comment (qili93) [inline paddle/fluid/framework/ir/ipu/transfer_cast_op_pass.cc]: 时间改为2022，下面其他新增文件页检查一下时间

File: paddle/fluid/framework/ir/ipu/ipu_pass_base.cc (hunk 1 above)
Comment (gglin001) [reply paddle/fluid/framework/ir/ipu/ipu_pass_base.cc]: fixed

File: paddle/fluid/framework/ir/ipu/transfer_cast_op_pass.cc (hunk 2 above)
Comment (gglin001) [reply paddle/fluid/framework/ir/ipu/transfer_cast_op_pass.cc]: fixed

**Review by qili93 — APPROVED** [review]
LGTM

File: paddle/fluid/framework/ir/ipu/optimizer_extract_pass.cc [hunk 3]
```diff
@@ -15,72 +15,301 @@
...
+  // square + reduce_sum + sum + sqrt + fill_constant
+  // + elementwise_max + elementwise_div + elementwise_mul
+  // clip_norm from fill_constant`s attr `value` dtype float
+
+  if (new_op.HasAttr("type")) {
+    auto new_node = graph->CreateOpNode(&new_op);
+    VLOG(10) << "New Optimizer Node:";
+    VLOG(10) << DebugString(new_node);
+  } else {
+    PADDLE_THROW(platform::errors::NotFound("No optimizer found"));
```
Comment (qili93) [inline paddle/fluid/framework/ir/ipu/optimizer_extract_pass.cc]: 根据CI提示，这里的error message不符合要求，参考以下文档进行修改

 https://github.com/PaddlePaddle/Paddle/wiki/Paddle-Error-Message-Writing-Specification

**Review by qili93 — CHANGES_REQUESTED** [review]

**Review by qili93 — APPROVED** [review]
LGTM

**Pull request merged**